godot-core/src/registry.rs

```diff
@@ -342,19 +458,42 @@ pub mod callbacks {
         create_custom(T::__godot_init)
     }
 
-    pub(crate) fn create_custom<T, F>(make_user_instance: F) -> sys::GDExtensionObjectPtr
+    #[cfg(since_api = "4.2")]
+    pub unsafe extern "C" fn recreate<T: cap::GodotInit>(
+        _class_userdata: *mut std::ffi::c_void,
+        object: sys::GDExtensionObjectPtr,
+    ) -> sys::GDExtensionClassInstancePtr {
+        create_rust_part_for_existing_godot_part(T::__godot_init, object)
+    }
+
+    pub(crate) fn create_custom<T, F>(make_user_instance: F) -> godot_ffi::GDExtensionObjectPtr
     where
         T: GodotClass,
         F: FnOnce(Base<T::Base>) -> T,
     {
-        let class_name = T::class_name();
         let base_class_name = T::Base::class_name();
 
-        //out!("create callback: {}", class_name.backing);
-
         let base_ptr =
             unsafe { interface_fn!(classdb_construct_object)(base_class_name.string_sys()) };
 
+        create_rust_part_for_existing_godot_part(make_user_instance, base_ptr);
+
+        // std::mem::forget(base_class_name);
+        base_ptr
+    }
+
+    fn create_rust_part_for_existing_godot_part<T, F>(
+        make_user_instance: F,
+        base_ptr: sys::GDExtensionObjectPtr,
+    ) -> sys::GDExtensionClassInstancePtr
```

Comment:
Thanks for your [elaboration in comment](https://github.com/godot-rust/gdext/pull/437#issuecomment-1741823263). I think this method would benefit from a short doc.

Reply:
^